runtime/src/lib.rs

```diff
@@ -101,6 +103,61 @@ impl Runtime {
     }
 }
 
+struct EnvAdapter<'a>{
+    system: &'a mut dyn System,
```

Comment:
This is exactly what I had in mind :+1:

What are your thoughts? Do you think it'll work?

Reply:
I _think_ it will work, but I guess it's quick to fill in some of the functions and actually test it.

Reply:
It works!